compiler/rustc_const_eval/src/interpret/memory.rs

```diff
@@ -706,10 +734,17 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
     /// Check whether an allocation is live. This is faster than calling
     /// [`InterpCx::get_alloc_info`] if all you need to check is whether the kind is
     /// [`AllocKind::Dead`] because it doesn't have to look up the type and layout of statics.
-    pub fn is_alloc_live(&self, id: AllocId) -> bool {
-        self.tcx.try_get_global_alloc(id).is_some()
-            || self.memory.alloc_map.contains_key_ref(&id)
-            || self.memory.extra_fn_ptr_map.contains_key(&id)
+    pub fn is_alloc_live(&self, id: AllocId) -> Liveness {
+        if self.memory.alloc_map.contains_key_ref(&id) {
+            return Liveness::Live;
+        }
```

Comment:
Note that these might be global. On their first read or write, globals get copied into the `alloc_map`, which is needed because we need to convert them from the global provenance type to that of Miri.

So what even is your definition of "global"? "AllocId is in `tcx`" or "AllocId is not in `memory`"? Those two are not equivalent.